File: array_strings/ts/1.ts
/*O(n)*/
const isUniqueWithHashTable = (word: string) => {
  const hashTable: Record<string, boolean> = {};
  for (const character of word) {
    if (hashTable[character.toLocaleLowerCase()]) return false;
    hashTable[character.toLocaleLowerCase()] = true;
  }
  return true;
};

console.log('aa (false): ', isUniqueWithHashTable('aa'));
console.log('aA (false): ', isUniqueWithHashTable('aA'));
console.log('ab (true): ', isUniqueWithHashTable('ab'));
console.log('aba (false): ', isUniqueWithHashTable('aba'));

File: array_strings/cpp/1.cpp
#include <iostream>
#include <map>
#include <utility>

bool isUniqueWithHashTable (std::string word) {
  std::map<char,int> hashTable;
  for(char character : word) {
    auto result = hashTable.emplace(character, 1);
    if (!result.second) return false;
  }
  return true;
}

int main() {
  std::string word = "uniqe";
  std::cout << isUniqueWithHashTable(word);
}